api/test/search.test.js

```diff
@@ -0,0 +1,21 @@
+import { default as driver } from "neo4j/driver";
+import { search } from "neo4j/queries/search";
+
+describe("search", () => {
+  afterAll(async () => {
+    await driver.close();
+  });
+
+  test("model search to have max 50 results per component type as default", async () => {
+    const data = await search({
+      searchTerm: "H2O",
+      model: "HumanGem",
+      version: HUMAN_GEM_VERSION,
+    });
+
+    expect(Object.keys(data)).toContain("Human-GEM");
+
+    const { metabolite } = data["Human-GEM"];
+    expect(metabolite.length).toBe(50);
```

Comment:
I don't quite understand why this test is failed. The limit is set to `50` in the `search` function. In addition, would it be better to change `toBe(50)` to `toBeLessThanOrEqual(50)`?

Reply:
Sorry, I should have pointed this out in the PR description. This is “supposed to” be failing right now. We have an actual bug in search at the moment. For metabolites results, we combine metabolites and compartmentalized metabolites before returning the results, where each node type has the limit of 50.

Reply:
I see. Thanks for the explanation! However, I think still `toBeLessThanOrEqual` might be better to interpret the expectation.

Reply:
Ah yes, I see your point. That would match better with the meaning of the test. The downside is that 0 results would also pass. I specifically wanted to test this on something that would push the limit to enforce we get 50 back, but maybe I can rewrite the description better or add a comment.

Reply:
Aha. I see what you mean. Metabolite for `H2O` has more than maximum hits. Then we should keep it but make a better description for this test.

Reply:
Added [a commit here](https://github.com/MetabolicAtlas/MetabolicAtlas/pull/830/commits/4b1671d0bb7c53e58ab4fd060e10c87f2e05dd84) to address it. Still sounds a bit awkward I think but do you think it's ok?

Reply:
How about "model search to have 50 results per component type if it exceeds the limit 50"?

Reply:
That sounds much better. Added another fix for it now. Thank you!